src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -175,28 +228,32 @@ private SwiftCompileStep makeCompileStep() {
               public String apply(SourcePath input) {
                 return getResolver().getAbsolutePath(input).toString();
               }
-            })));
+            })
+            .toSet()));
 
     Optional<Iterable<String>> configFlags = swiftBuckConfig.getFlags();
     if (configFlags.isPresent()) {
       compilerCommand.addAll(configFlags.get());
     }
-    compilerCommand.add(
-        "-enable-testing",
-        "-c",
-        enableObjcInterop ? "-enable-objc-interop" : "",
-        hasMainEntry ? "" : "-parse-as-library",
-        "-module-name",
-        moduleName,
-        "-emit-module",
-        "-emit-module-path",
-        modulePath.toString(),
-        "-o",
-        objectPath.toString(),
-        "-emit-objc-header-path",
-        headerPath.toString());
-    for (SourcePath sourcePath : srcs) {
-      compilerCommand.add(getResolver().getRelativePath(sourcePath).toString());
+
+    if (bridgingHeader.isPresent()) {
+      compilerCommand.add(
+          "-import-objc-header",
+          getResolver().getRelativePath(bridgingHeader.get()).toString());
+    }
+
+    if (src.isPresent()) {
+      compilerCommand.add(
+          "-primary-file",
+          getResolver().getAbsolutePath(src.get()).toString(),
+          "-emit-module-path",
+          modulePath.toString(),
+          "-o",
+          objectPath.toString());
+    } else {
+      compilerCommand.add(
+          "-o",
+          modulePath.toString());
```

Comment:
When merging all single file outputs, the output is another swift module, not an object file.